tf_agents/agents/ddpg/critic_rnn_network.py

```diff
@@ -62,136 +72,97 @@ def __init__(self,
         each item is the number of units in the layer. This is applied after the
         LSTM cell.
       activation_fn: Activation function, e.g. tf.nn.relu, slim.leaky_relu, ...
+      dtype: The dtype to use by the convolution and fully connected layers.
       name: A string representing name of the network.
 
     Returns:
       A tf.float32 Tensor of q-values.
 
     Raises:
-      ValueError: If `observation_spec` or `action_spec` contains more than one
+      ValueError: If `action_spec` contains more than one
         item.
     """
     observation_spec, action_spec = input_tensor_spec
 
-    if len(tf.nest.flatten(observation_spec)) > 1:
-      raise ValueError(
-          'Only a single observation is supported by this network.')
+    kernel_initializer = tf.compat.v1.variance_scaling_initializer(
+      scale=2.0, mode='fan_in', distribution='truncated_normal')
+
+    obs_encoder = encoding_network.EncodingNetwork(
+        observation_spec,
+        preprocessing_layers=observation_preprocessing_layers,
+        preprocessing_combiner=observation_preprocessing_combiner,
+        conv_layer_params=observation_conv_layer_params,
+        fc_layer_params=observation_fc_layer_params,
+        activation_fn=activation_fn,
+        kernel_initializer=kernel_initializer,
+        dtype=dtype,
+        name='obs_encoding')
 
     if len(tf.nest.flatten(action_spec)) > 1:
       raise ValueError('Only a single action is supported by this network.')
 
-    observation_layers = utils.mlp_layers(
-        observation_conv_layer_params,
-        observation_fc_layer_params,
-        activation_fn=activation_fn,
-        kernel_initializer=tf.compat.v1.keras.initializers.VarianceScaling(
-            scale=1. / 3., mode='fan_in', distribution='uniform'),
-        name='observation_encoding')
-
-    action_layers = utils.mlp_layers(
+    action_layers = tf.keras.Sequential(utils.mlp_layers(
         None,
         action_fc_layer_params,
         activation_fn=activation_fn,
         kernel_initializer=tf.compat.v1.keras.initializers.VarianceScaling(
             scale=1. / 3., mode='fan_in', distribution='uniform'),
-        name='action_encoding')
-
-    joint_layers = utils.mlp_layers(
-        None,
-        joint_fc_layer_params,
+        name='action_encoding'))
+
+    obs_encoding_spec = tf.TensorSpec(
+        shape=(observation_fc_layer_params[-1],), dtype=tf.float32)
+    lstm_encoder = lstm_encoding_network.LSTMEncodingNetwork(
+        input_tensor_spec=(obs_encoding_spec, action_spec),
```

Comment:
can you add a comment explaining why you're encoding the action with observation via the LSTM encoder here?  something seems off about this.  then again, i'm unfamiliar with this part of the code.  @oars any idea?